File: app/src/main/java/com/my/televip/virtuals/messenger/FileLoader.java
package com.my.televip.virtuals.messenger;

import com.my.televip.Class.ClassNames;
import com.my.televip.Class.ClassLoad;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import java.io.File;

import de.robv.android.xposed.XposedHelpers;

public class FileLoader {

    Object fileLoader;

    public FileLoader(Object fileLoader){
        this.fileLoader = fileLoader;
    }

    public File getLocalFile(ImageLocation location) {
        return (File) XposedHelpers.callMethod(fileLoader, Obfuscate.getMethodName("FileLoader", "getLocalFile"), location.imageLocation);
    }

    public static FileLoader getInstance(int num) {
        return new FileLoader(XposedHelpers.callStaticMethod(ClassLoad.getClass(ClassNames.FILE_LOADER), Obfuscate.getMethodName("FileLoader", "getInstance"), num));
    }

}

File: app/src/main/java/com/my/televip/virtuals/messenger/FileLoader.kt
package com.my.televip.virtuals.messenger

import com.my.televip.clazz.ClassLoad
import com.my.televip.clazz.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class FileLoader(@JvmField val fileLoader: Any?) {

    companion object {
        @JvmStatic
        fun checkFileName(fileName: String?): String? {
            val fileLoaderClass = ClassLoad.getClass(ClassNames.FILE_LOADER) ?: return fileName
            return XposedHelpers.callStaticMethod(
                fileLoaderClass,
                Obfuscate.getMethodName("FileLoader", "checkFileName"),
                fileName
            ) as? String
        }
    }
}
